openmsupply/open-msupply
# 2023 mutate patient without program config

Fixes #2023 

# 👩🏻‍💻 What does this PR do? 
- Rename existing mutate patient endpoints to `insertProgramPatient` and `updateProgramPatient`
- Introduce new `insertPatient` and `updatePatient` endpoints to mutate a patient without updating a document
- Update FE to use these new endpoints when no patient registry is found
- When there is not patient registry, the FE uses some minimal hardecoded json schema to create/edit a patient. This schema maps directly to fields in the name table

Note, the UI is currently not handling missing patient mutation permission in mSupply "Edit patient details". While it correctly not allow to mutate a patient it displays a lot of errors along the way...

# Testing:
1) Use a mSupply data file without programs config and connect to omSupply
- Patient creation and editing should work with a minimal json schema
- After editing a patient changes should be stored in the name table but there shouldn't be any documents
- Test that new users sync

2) Use data file with program config
- Creating an mutating a patient should still work as before, i.e. documents should be created
